lib/rollbar/delay/thread.rb

```diff
@@ -1,9 +1,61 @@
+require 'thread'
+require 'timeout'
+
 module Rollbar
   module Delay
     class Thread
-      def self.call(payload)
-        new.call(payload)
-      end
+      Error        = Class.new(StandardError)
+      TimeoutError = Class.new(Error)
+
+      class << self
+        def call(payload)
+          spawn_threads_reaper
+          threads << new.call(payload)
+        end
+
+        private
+
+        def threads
+          @threads ||= Queue.new
+        end
+
+        def reaper
+          @reaper ||=
+            Thread.start do
+              loop do
+                thread = threads.pop
+
+                if thread == :exit
+                  break
+                else
+                  thread.join
+                end
+              end
+            end
+        end
+
+        def spawn_threads_reaper
+          unless @spawned
```

Comment:
According to the Ruby style, it's prefered not wrap a method inside a `unless` block. Here you can do `return if @spawned`